packages/ui/src/models/visualization/visualization-node.test.ts

```diff
@@ -167,6 +168,37 @@ describe('VisualizationNode', () => {
     expect(child.getParentNode()).toEqual(node);
   });
 
+  it('should add a nodeInteraction', () => {
+    const TestNodeInteraction: NodeInteraction = {
+      canHavePreviousStep: true,
+      canHaveNextStep: true,
+      canHaveChildren: true,
+      canHaveSpecialChildren: true,
+      canReplaceStep: false,
+      canRemoveStep: false,
+      canRemoveFlow: false,
+      canBeDisabled: true,
+    };
+    node.addNodeInteraction(TestNodeInteraction);
+
+    expect(node.getNodeInteraction()).toEqual(TestNodeInteraction);
+  });
+
+  describe('getNodeInteraction', () => {
+    it('should return DISABLED_NODE_INTERACTION', () => {
+      expect(node.getNodeInteraction()).toEqual({
+        canHavePreviousStep: false,
+        canHaveNextStep: false,
+        canHaveChildren: false,
+        canHaveSpecialChildren: false,
+        canReplaceStep: false,
+        canRemoveStep: false,
+        canRemoveFlow: false,
+        canBeDisabled: false,
+      });
+    });
+  });
+
```

Comment:
@shivamG640 , I think we need to update these tests a bit please :pray: :

1. We're using `describe` just to wrap a single test, let's try to wrap all `getNodeInteraction` tests instead.
2. We're testing the first step of the return statement, that provided a `nodeInteraction`, this one is returned :heavy_check_mark: 
3. We're not testing what happens when there's a base entity and if we return that node interaction instead.
4. We're indirectly testing that we return `DISABLED_NODE_INTERACTION` when there's no base entity, but I think we can make it more explicitly by either setting the baseEntity to `undefined` or similar, remember, if the test fails, we want to know exactly and as fast as possible why it failed, so explicitness is key.
5. For the last one, we could compare against the `DISABLED_NODE_INTERACTION`, this way, if we update the object, we don't need to edit it in the test itself.

Lastly, **and this is totally up to you**, if you want to reduce the number of lines, you could replace the `toEqual()` method with `toMatchSnapshot()`, this way you'll get the same validation but it will be placed on the snapshot file.

```suggestion
  describe("getNodeInteraction", () => {
    it("should add a nodeInteraction", () => {
      const testNodeInteraction: NodeInteraction = {
        canHavePreviousStep: true,
        canHaveNextStep: true,
        canHaveChildren: true,
        canHaveSpecialChildren: true,
        canReplaceStep: false,
        canRemoveStep: false,
        canRemoveFlow: false,
        canBeDisabled: true,
      };
      node.addNodeInteraction(testNodeInteraction);

      expect(node.getNodeInteraction()).toEqual(testNodeInteraction);
    });

    it("should return node interaction from the base entity and ensure it was called", () => {
      // explicitly set the baseEntity to something
      // check the baseEntity's getNodeInteraction method was called
      expect(node.getNodeInteraction()).toEqual({
        canHavePreviousStep: false,
        canHaveNextStep: false,
        canHaveChildren: false,
        canHaveSpecialChildren: false,
        canReplaceStep: false,
        canRemoveStep: false,
        canRemoveFlow: false,
        canBeDisabled: false,
      });
    });

    it("should return DISABLED_NODE_INTERACTION when there is no underlying base entity", () => {
      // explicitly set the baseEntity to undefined
      expect(node.getNodeInteraction()).toEqual({
        canHavePreviousStep: false,
        canHaveNextStep: false,
        canHaveChildren: false,
        canHaveSpecialChildren: false,
        canReplaceStep: false,
        canRemoveStep: false,
        canRemoveFlow: false,
        canBeDisabled: false,
      });
    });
  });
```